frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditDataType/SelectDataType.tsx

```diff
@@ -21,33 +21,40 @@ export const SelectDataType = ({
 }: SelectDataTypeProps) => {
   const { t } = useTranslation();
   const { mutateDataType } = useBpmnApiContext();
-  const handleChangeDataModel = (dataModelId?: string) => {
-    if (dataModelId === existingDataType) return;
-    const dataTypeChange: DataTypeChange = {
-      newDataType: dataModelId,
-      connectedTaskId: connectedTaskId,
-    };
-    mutateDataType(dataTypeChange);
+  const currentValue = existingDataType ? [existingDataType] : [];
+  const handleChangeDataModel = (newDataModelIds?: string[]) => {
+    const newDataModelId = newDataModelIds ? newDataModelIds[0] : undefined;
+    if (newDataModelId !== existingDataType) {
+      const dataTypeChange: DataTypeChange = {
+        newDataType: newDataModelId,
+        connectedTaskId: connectedTaskId,
+      };
+      mutateDataType(dataTypeChange);
+    }
     onClose();
   };
 
   return (
     <div className={classes.dataTypeSelect}>
-      <NativeSelect
-        size='small'
-        onChange={({ target }) => handleChangeDataModel(target.value)}
+      <Combobox
         label={t('process_editor.configuration_panel_set_datamodel')}
-        value={existingDataType ?? 'noModelKey'}
+        inputValue={currentValue[0]}
```

Comment:
Is this supposed to always be the first value? 
What happens when we have currentValue set to []? 😄

Reply:
It will show nothing which is the actual case when there is no existing data type, but I figured out that it is not needed since value is shown as the input anyway 😁
But nice that you questioned it!